mlir/include/mlir/IR/OperationSupport.h

```diff
@@ -668,6 +674,13 @@ class RegisteredOperationName : public OperationName {
 
       return {};
     }
+    void printProperties(Operation *op, OpAsmPrinter &printer) final {
+      if constexpr (hasProperties) {
+        auto concreteOp = cast<ConcreteOp>(op);
+        ConcreteOp::printProperties(concreteOp->getContext(), printer,
+                                    concreteOp.getProperties());
```

Comment:
self-review: perhaps I should be explicit here to also pass `ArrayRef<StringRef>{}` as the 4th argument?